phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -326,6 +326,49 @@ private IndexMaintainer(PTable dataTable, PTable index) {
         // TODO: check whether index is immutable or not. Currently it's always false so checking
         // data table is with immutable rows or not.
         this.immutableRows = dataTable.isImmutableRows();
+        int indexColByteSize = 0;
+        for (int i = indexPosOffset; i < index.getPKColumns().size(); i++) {
+            PColumn indexColumn = index.getPKColumns().get(i);
+            int indexPos = i - indexPosOffset;
+            Expression expression = null;
+            try {
+                ParseNode parseNode  = SQLParser.parseCondition(indexColumn.getExpressionStr());
```

Comment:
Also, this is where you'd want a SQLParser that makes all column, table, and schema references as case sensitive (i.e. for ColumnParseNode). You could just have a parseCaseSensitiveExpression static method.